src/Raven.Server/Documents/Indexes/Static/CurrentIndexingScope.cs

```diff
@@ -222,7 +232,21 @@ public unsafe dynamic LoadDocument(LazyStringValue keyLazy, string keyString, st
                 if (document.TryGetMetadata(out var metadata) && metadata.TryGet(Constants.Documents.Metadata.Collection, out string collection))
                 {
                     if (string.Equals(collection, collectionName, StringComparison.OrdinalIgnoreCase) == false)
+                    {
+                        if (_mismatchedReferencesHandler.MismatchedReferences.Count < MismatchedReferencesHandler.MaxMismatchedDocumentLoadsPerIndex)
```

Comment:
Maybe introduce a method or property in `MismatchedReferencesHandler` which checks this condition under the covers e.g. `IsFull` or `CanHandleMoreDocuments`. This way you don't need to expose `MismatchedReferencesHandler` at all.

Reply:
Done